unit test failure: persistent volume controller TestFindingPreboundVolumes

/kind bug
cc @kubernetes/sig-storage-bugs 

Hit this in https://github.com/kubernetes/kubernetes/pull/55843

https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/pr-logs/pull/55843/pull-kubernetes-bazel-test/18828/

```
E1121 18:57:34.775318      13 goroutinemap.go:165] Operation for "delete-volume8-3[]" failed. No retries permitted until 2017-11-21 18:57:35.275153547 +0000 UTC m=+11.944865534 (durationBeforeRetry 500ms). Error: "Error getting deleter volume plugin for volume \"volume8-3\": no volume plugin matched"
E1121 18:57:34.777969      13 goroutinemap.go:165] Operation for "delete-volume8-4[]" failed. No retries permitted until 2017-11-21 18:57:35.277895529 +0000 UTC m=+11.947607508 (durationBeforeRetry 500ms). Error: "Failed to create deleter for volume \"volume8-4\": Mock plugin error: no deleteCalls configured"
E1121 18:57:34.779201      13 goroutinemap.go:165] Operation for "delete-volume8-5[]" failed. No retries permitted until 2017-11-21 18:57:35.27915186 +0000 UTC m=+11.948863825 (durationBeforeRetry 500ms). Error: "Mock delete error"
E1121 18:57:34.800542      13 goroutinemap.go:165] Operation for "delete-volume9-1[]" failed. No retries permitted until 2017-11-21 18:57:35.300444299 +0000 UTC m=+11.970156259 (durationBeforeRetry 500ms). Error: "Mock delete error"
--- FAIL: TestFindingPreboundVolumes (0.00s)
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x20 pc=0x19056d2]

goroutine 49 [running]:
testing.tRunner.func1(0xc4203683c0)
	/root/.cache/bazel/_bazel_root/e9f728bbd90b3fba632eb31b20e1dacd/external/go_sdk/src/testing/testing.go:711 +0x5d9
panic(0x1ab3fe0, 0x2672fa0)
	/root/.cache/bazel/_bazel_root/e9f728bbd90b3fba632eb31b20e1dacd/external/go_sdk/src/runtime/panic.go:491 +0x2a2
k8s.io/kubernetes/pkg/controller/volume/persistentvolume.TestFindingPreboundVolumes(0xc4203683c0)
	pkg/controller/volume/persistentvolume/index_test.go:971 +0xb22
testing.tRunner(0xc4203683c0, 0x1caedf8)
	/root/.cache/bazel/_bazel_root/e9f728bbd90b3fba632eb31b20e1dacd/external/go_sdk/src/testing/testing.go:746 +0x16d
created by testing.(*T).Run
	/root/.cache/bazel/_bazel_root/e9f728bbd90b3fba632eb31b20e1dacd/external/go_sdk/src/testing/testing.go:789 +0x569
```

I can reproduce by stress testing the tests.

```
$ go get -u golang.org/x/tools/cmd/stress
$ go test -o persistentvolume.test -i -v k8s.io/kubernetes/pkg/controller/volume/persistentvolume
$ stress ./persistentvolume.test -test.v
/tmp/go-stress294112895                                    
=== RUN   TestSync                                         
--- PASS: TestSync (0.02s)                                 
=== RUN   TestSyncAlphaBlockVolume                         
--- PASS: TestSyncAlphaBlockVolume (0.00s)                 
=== RUN   TestMultiSync                                    
--- PASS: TestMultiSync (10.59s)                           
=== RUN   TestDeleteSync                                   
E1121 11:42:31.688175   15970 goroutinemap.go:165] Operation for "delete-volume8-3[]" failed. No retries permitted until 2017-11-21 11:42:32.187770283 -0800 PST m=+11.177179646 (durationBeforeRetry 500ms). Error: "Error getting deleter volume plugin for volume \"volume8-3\": no volume plugin matched"
E1121 11:42:31.690779   15970 goroutinemap.go:165] Operation for "delete-volume8-4[]" failed. No retries permitted until 2017-11-21 11:42:32.190715121 -0800 PST m=+11.180124565 (durationBeforeRetry 500ms). Error: "Failed to create deleter for volume \"volume8-4\": Mock plugin error: no deleteCalls configured"
E1121 11:42:31.691689   15970 goroutinemap.go:165] Operation for "delete-volume8-5[]" failed. No retries permitted until 2017-11-21 11:42:32.191639007 -0800 PST m=+11.181048313 (durationBeforeRetry 500ms). Error: "Mock delete error"      
--- PASS: TestDeleteSync (0.03s)                           
=== RUN   TestDeleteMultiSync                              
E1121 11:42:31.713821   15970 goroutinemap.go:165] Operation for "delete-volume9-1[]" failed. No retries permitted until 2017-11-21 11:42:32.21380399 -0800 PST m=+11.203213227 (durationBeforeRetry 500ms). Error: "Mock delete error"       
--- PASS: TestDeleteMultiSync (1.88s)                      
=== RUN   TestMatchVolume                                  
--- PASS: TestMatchVolume (0.00s)                          
=== RUN   TestMatchingWithBoundVolumes                     
--- PASS: TestMatchingWithBoundVolumes (0.00s)             
=== RUN   TestListByAccessModes                            
--- PASS: TestListByAccessModes (0.00s)                    
=== RUN   TestAllPossibleAccessModes                       
--- PASS: TestAllPossibleAccessModes (0.00s)               
=== RUN   TestFindingVolumeWithDifferentAccessModes        
--- PASS: TestFindingVolumeWithDifferentAccessModes (0.00s)                                                            
=== RUN   TestAlphaVolumeModeCheck                         
--- PASS: TestAlphaVolumeModeCheck (0.00s)                 
=== RUN   TestAlphaFilteringVolumeModes                    
--- PASS: TestAlphaFilteringVolumeModes (0.00s)            
=== RUN   TestFindingPreboundVolumes                       
--- FAIL: TestFindingPreboundVolumes (0.00s)               
pan                                                        
6 runs so far, 3 failures
```

Interestingly running `TestFindingPreboundVolumes` alone doesn't cause the panic. It looks like something caused by interaction with the other tests.

```
$ stress ./persistentvolume.test -test.v -test.run TestFindingPreboundVolumes
145 runs so far, 0 failures
286 runs so far, 0 failures
```

--- a/pkg/controller/volume/persistentvolume/binder_test.go
+++ b/pkg/controller/volume/persistentvolume/binder_test.go
@@ -720,14 +720,9 @@ func TestSyncAlphaBlockVolume(t *testing.T) {
 		t.Errorf("Failed to enable feature gate for BlockVolume: %v", err)
 		return
 	}
+	defer utilfeature.DefaultFeatureGate.Set("BlockVolume=false")
 
 	runSyncTests(t, tests, []*storage.StorageClass{})
-
-	err1 := utilfeature.DefaultFeatureGate.Set("BlockVolume=false")
-	if err1 != nil {
-		t.Errorf("Failed to disable feature gate for BlockVolume: %v", err)
-		return
-	}
 }
 
 // Test multiple calls to syncClaim/syncVolume and periodic sync of all

--- a/pkg/controller/volume/persistentvolume/index_test.go
+++ b/pkg/controller/volume/persistentvolume/index_test.go
@@ -839,6 +839,8 @@ func TestAlphaVolumeModeCheck(t *testing.T) {
 		}
 	}
 
+	// make sure feature gate is turned off
+	toggleBlockVolumeFeature(false, t)
 }
 
 func TestAlphaFilteringVolumeModes(t *testing.T) {
@@ -933,6 +935,9 @@ func TestAlphaFilteringVolumeModes(t *testing.T) {
 			t.Errorf("Unexpected failure for scenario: %s - %+v", name, err)
 		}
 	}
+
+	// make sure feature gate is turned off
+	toggleBlockVolumeFeature(false, t)
 }
 
 func TestFindingPreboundVolumes(t *testing.T) {

